### PR TITLE
Check if context is NULL

### PR DESCRIPTION
**Pull request by Bomb3rman**

### CODE REVIEW

**Comment by bachonk** [comment]
Thanks @Bomb3rman! Could you provide a way to reproduce a case where `context = NULL` so I can verify this patch?

**Comment by bachonk** [comment]
@Bomb3rman looks like some interesting stuff! Could you summarize your changes in the description of your PR?

**Comment by Bomb3rman** [comment]
Sorry, I did not really wanna create a pull request for that. Just some stuff we use internally. I might actually clean this up and push it at a later stage.
